feat: add screenshots

Moderndo opened the pull request:
This pull request adds screenshots for the backend setup and testing.

Changes include:
- Added screenshots under docs/screenshots/ showing backend setup and test results.
- Verified backend functionality with Invoke-RestMethod test returning 200 OK.
- Updated .gitignore and related backend files for Windows-friendly run steps.

These screenshots and updates will help document and verify the backend configuration process for future reference.

Moderndo committed:
feat: add screenshots

Moderndo committed:
Resolve merge: prefer feature versions of package.json and src/index.js

Ed-ward-Harris commented:
Hi Manaal 

Could for the ai feautures you have implemented could you please a detailed comment of what you have added and how it works. As it stands your comment only describes the screenshots you have added to the docs. Thanks

Moderndo commented:
Hi Edward,

This is the comment you asked me for regarding the AI features I implemented. I hope this meets your requirements.

AI Features Implemented

1. Chat Screen Functionality

Welcome Messages
When the chat screen loads for the first time, it automatically seeds three welcome messages from the bot to guide the user.

Message Bubbles

User messages appear on the right side, styled with the user bubble.

Bot replies appear on the left side, styled with the bot bubble.
This makes the conversation visually clear and consistent.

Predefined FAQ Replies
Common user queries are handled directly without calling the backend.
Examples:

“What is this app?” → “This is a Smishing Detection app that helps you identify and report SMS phishing.”

“How do I report?” → “You can report suspicious SMS in the Community Report tab.”

“What is smishing?” → “Smishing is SMS-based phishing…”

Escalation to Ollama Backend
If the input does not match an FAQ, the message is forwarded to the backend (Flask API with Ollama). After a few support prompts, the system escalates automatically to the AI model for deeper responses.
This is the Youtube video Link:
https://youtu.be/DOwg_YxFoxo
Thanks

Ed-ward-Harris commented:
Hi Manaal thanks for the PR

It looks like the chat bot works as intended. Although there are some issues that need to be resolved.

### 1 The index.js file has been gutted. 
You have removed:
- The security middleware
- rate limiting (global and auth rate limiter)
- routes (scanRoute, 
- imports

These all need to re-added to the index.js file. As is the app losses protections and features and endpoints will be broken.

### 2. Problems with the Flask service
- You have hard coded the configs (ip, port etc) but refer to a .env file in the readme. These should be in an env file.
-  instead of creating a new .env you can add fields to the existing backend .env file.
- The service runs in debug mode (debug=True) on all interfaces (0.0.0.0) there is no auth, no rate limiting, now input validation. This would be okay for a testing environment, but as it stands wouldn't be safe to use in production.
- The readme instructions are slightly wrong, please double check them. They also refer to a a health end point that doesn't seem to exist.

Moderndo committed:
backend: restore security & rate limiting; env-driven Flask proxy; add health endpoints; docs

Moderndo committed:
backend fixes: env configs, chat proxy, readmes

Moderndo commented:
Hi Edward, thanks for the detailed feedback.
I’ve restored the index.js so that security middleware, global and auth rate limiting, and all route imports (scanRoute, chatRoute, contactRoute, spamRoute, authRoute) are back in place.

On the Flask side, I’ve removed hardcoded configs and moved them into environment variables as suggested. The .env.example has been updated accordingly. I also adjusted the README instructions so they now match the actual setup, including the correct health endpoint and environment-driven configuration.

<img width="1377" height="697" alt="Screenshot 2025-09-13 131737" src="https://github.com/user-attachments/assets/cd5992d1-3c93-40aa-9cae-d3ae8343980c" />

<img width="1064" height="805" alt="Screenshot 2025-09-13 131809" src="https://github.com/user-attachments/assets/12cfd85d-ae6f-4405-8e91-29c781c15960" />
<img width="1087" height="869" alt="Screenshot 2025-09-13 132458" src="https://github.com/user-attachments/assets/54186dcf-0fbf-430e-a310-bc84409e3a7d" />
<img width="1174" height="388" alt="Screenshot 2025-09-13 132601" src="https://github.com/user-attachments/assets/0fe5a1de-12ef-440a-b986-ca5dfca4ef43" />

<img width="1092" height="855" alt="Screenshot 2025-09-13 182136" src="https://github.com/user-attachments/assets/eea55926-eb21-4aff-80e7-69e7607ce4f2" />
<img width="1159" height="501" alt="Screenshot 2025-09-13 182749" src="https://github.com/user-attachments/assets/49eb3fdf-1317-4067-ae8c-3d27c6edb575" />

Moderndo committed:
Merge branch 'dev' into feature/ai-chat-api

Ed-ward-Harris commented:
Thanks for the update. You have removed the file structure from the README, you have removed the original variables from the .ENV file, it currently only has the ones that you added. Please simply restore the index.js page to its original state, then add only what is needed for your app to function. One these are addressed I will be able to merge.

Moderndo committed:
Merge branch 'dev' into feature/ai-chat-api

Moderndo committed:
Merge branch 'dev' into feature/ai-chat-api

Moderndo committed:
fix: restore README and revert security middleware

Moderndo committed:
fix: restore root README to original state

Moderndo committed:
fix: revert rate limiter middleware to original state

Moderndo committed:
refactor: clean up index.js (removed duplicates, structured middlewares/routes)

Moderndo committed:
Update index.js

Moderndo committed:
Update index.js

Ed-ward-Harris committed:
Update index.js
